INSTALLATION.rst

```diff
@@ -61,59 +63,89 @@ Recommended Method
 
    In addition to the default packages, many other developers provide
    "channels" to distribute their own builds of a variety of software.
-   These days, the most useful channel is `conda-forge`, which provides
+   These days, the most useful channel is conda-forge_, which provides
    everything necessary to install :term:`FiPy`.
 
 * `install Miniconda`_ on your computer
-* run::
+* Install :term:`FiPy` :ref:`REQUIREDPACKAGES` with one of the following:
 
-    $ conda create --name <MYFIPYENV> --channel conda-forge python=<PYTHONVERSION> fipy gmsh
+    * Conda environment files::
 
-  .. note::
+        $ conda env create --name <MYFIPYENV> \
+            --file environments/base-environment.yml
 
-     This command creates a self-contained conda_ environment and then
-     downloads and populates the environment with the prerequisites for
-     :term:`FiPy` from the conda-forge_ channel at https://anaconda.org.
+      followed by::
 
-     :term:`Gmsh` is an optional package because some versions are
-     incompatible with :term:`FiPy`, so it must be requested explicitly.
+        $ conda env update --name <MYFIPYENV> \
+            --file environments/<SOLVER>-environment.yml
 
-  .. note::
+      You can try to include multiple solver suites, but be aware that
+      different suites may have incompatible requirements, or may restrict
+      installation to obsolete versions of Python.  Given that :term:`FiPy`
+      can only use one solver suite during a run, installing more than one
+      solver in an environment isn't necessary.
+
+      .. attention::
+
+         Successively updating an environment can be unpredictable, as
+         later packages may conflict with earlier ones.  Unfortunately,
+         ``conda env create`` `does not support multiple environment files
+         <https://github.com/conda/conda/issues/9294>`_.
+
+         Alternatively, combine :file:`environments/base-environment.yml`
+         with any :file:`environments/<SOLVER>-environment.yml` you wish to
+         use (`conda-merge <https://github.com/amitbeka/conda-merge>`_ may
+         prove useful).  Then execute::
 
-     The `fipy conda-forge`_ package is a convenience. You may choose to
-     install packages explicitly, e.g.,::
+           $ conda env create --name <MYFIPYENV> --file <MYMERGEDENVIRONMENT>.yml
 
-       $ conda create --name <MYFIPYENV> --channel conda-forge python=3 numpy scipy matplotlib-base future packaging mpich mpi4py petsc4py mayavi "gmsh <4.0|>=4.5.2"
+    * `conda-lock <https://github.com/conda/conda-lock>`_ lockfiles::
```

Comment:
This option will pin all the packages so is the most reproducible, but not particularly upgradable. For most, this is the safest way to generate a FiPy environment that actually works.